libraries/AP_RangeFinder/AP_RangeFinder.cpp

```diff
@@ -602,12 +609,13 @@ bool RangeFinder::has_orientation(enum Rotation orientation) const
 // find first range finder instance with the specified orientation
 AP_RangeFinder_Backend *RangeFinder::find_instance(enum Rotation orientation) const
 {
-    // first try for a rangefinder that is in range
+    // first try for a rangefinder that is in range and also in use
     for (uint8_t i=0; i<num_instances; i++) {
         AP_RangeFinder_Backend *backend = get_backend(i);
         if (backend != nullptr &&
             backend->orientation() == orientation &&
-            backend->status() == Status::Good) {
+            backend->status() == Status::Good &&
+            backend->use_sensor()) {
             return backend;
```

Comment:
need to check all uses of get_backend() to ensure they obey use setting
eg: NavEKF3_core::readRangeFinder()
also AP_Proximity_RangeFinder::update